exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/JoinStatus.java

```diff
@@ -81,114 +58,82 @@ public JoinStatus(RecordBatch left, RecordBatch right, MergeJoinBatch output) {
   @Override
   public String toString() {
     return
-        super.toString()
+      super.toString()
         + "["
-        + "leftPosition = " + leftPosition
-        + ", rightPosition = " + rightPosition
-        + ", svRightPosition = " + svRightPosition
+        + "leftPosition = " + left.getCurrentPosition()
+        + ", rightPosition = " + right.getCurrentPosition()
         + ", outputPosition = " + outputPosition
-        + ", lastLeft = " + lastLeft
-        + ", lastRight = " + lastRight
-        + ", rightSourceMode = " + rightSourceMode
-        + ", sv4 = " + sv4
         + ", joinType = " + joinType
         + ", ok = " + ok
         + ", initialSet = " + initialSet
-        + ", leftRepeating = " + leftRepeating
         + ", left = " + left
         + ", right = " + right
         + ", outputBatch = " + outputBatch
         + "]";
   }
 
-  public boolean hasIntermediateData() {
-    return hasIntermediateData;
-  }
-
-  public void resetIntermediateData() {
-    hasIntermediateData = false;
-  }
-
-  public void setIntermediateData(int initialRightPosition, boolean crossedBatchBoundaries) {
-    this.initialRightPosition = initialRightPosition;
-    this.crossedBatchBoundaries = crossedBatchBoundaries;
-    this.hasIntermediateData = true;
-  }
-
-  public int getInitialRightPosition() {
-    return initialRightPosition;
-  }
-
-  public boolean getCrossedBatchBoundaries() {
-    return crossedBatchBoundaries;
-  }
-
-  private final IterOutcome nextLeft() {
-    return outputBatch.next(LEFT_INPUT, left);
-  }
-
-  private final IterOutcome nextRight() {
-    return outputBatch.next(RIGHT_INPUT, right);
-  }
 
   public final void ensureInitial() {
     switch(initialSet) {
       case INIT:
-        this.lastLeft = nextLeft();
-        this.lastRight = nextRight();
+        advanceLeft();
+        advanceRight();
         initialSet = InitState.CHECK;
         break;
       case CHECK:
-        if (lastLeft != IterOutcome.NONE && left.getRecordCount() == 0) {
-          this.lastLeft = nextLeft();
+        if (getLeftStatus() != IterOutcome.NONE && left.getInnerRecordCount() == 0) {
```

Comment:
can you add comments on this? Both the method and these checks. What happens if the iteroutcome is something like STOP?